internal/sgbouncewizard/sgbouncewizard.go

```
@@ -0,0 +1,237 @@
+	var rule models.BounceRule
+	
+	err := decoder.Decode(&rule)
+	
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	data, err := srv.DBClient.CreateRuleDB(&rule)
+	w.Write(data)
+}
+
+func (srv *Server) updateRule(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	prevRule := *r.Context().Value("rule").(*models.BounceRule)
+	decoder := json.NewDecoder(r.Body)
+	var newRule models.BounceRule
+	err := decoder.Decode(&newRule)
+
+	if err != nil {
+		fmt.Println("error her?")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	data, err := json.Marshal(newRule)
+	ruleDifferences := db.GetRuleDifferences(&prevRule, &newRule)
+	srv.DBClient.UpdateRuleDB(ruleDifferences, &prevRule)
+	w.Write(data)
+}
+
+
+func (srv *Server) GetChangelog(w http.ResponseWriter, r *http.Request) {
+	rules, err := srv.DBClient.Changelog()
+
+	data, err := json.Marshal(*rules)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.WriteHeader(200)
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(data)
+}
+
+func (srv *Server) Serve() {
+	r := chi.NewRouter()
+
+	r.Use(middleware.RequestID)
+	r.Use(middleware.RealIP)
+	r.Use(middleware.Logger)
+	r.Use(middleware.Recoverer)
+
+	r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("welcome"))
+	})
+
+	r.Route("/user", func(r chi.Router) {
+		r.Post("/", srv.CheckUser)
+	})
+
+	r.Route("/changelogs", func(r chi.Router) {
+		r.Get("/", srv.GetChangelog)
+	})
+
+	r.Route("/bounce_rules", func(r chi.Router) {
+		r.Get("/", srv.ListRules)
+		r.Post("/", srv.createRule)
+
+		r.Route("/{bounce_id}", func(r chi.Router) {
+			r.Use(srv.RuleContext)
+			r.Get("/", srv.getRule)
+			r.Delete("/", srv.deleteRule)
+			r.Put("/", srv.updateRule)
+		})
+	})
+
+	http.ListenAndServe(":3000", r)
```

This returns an error that is being ignored. If your server can't bind to port 3000 for example then the app will appear to just terminate. You should return the error to the caller where it should be handled appropriately